pkg/controller/seed-controller-manager/mla/resources.go

```diff
@@ -275,7 +302,17 @@ func GatewayDeploymentCreator(data *resources.TemplateData) reconciling.NamedDep
 				RunAsUser:    pointer.Int64Ptr(1001),
 				RunAsNonRoot: pointer.BoolPtr(true),
 			}
-			d.Spec.Template.Labels = d.Spec.Selector.MatchLabels
+			// hash for the annotation used to force pod restart upon configuration change
+			configHash := sha1.New()
+			configData, err := json.Marshal(settings)
+			if err != nil {
+				return nil, fmt.Errorf("failed to encode MLAAdminSetting: %v", err)
+			}
+			configHash.Write(configData)
+			d.Spec.Template.Labels = map[string]string{
+				configHashAnnotation: fmt.Sprintf("%x", configHash.Sum(nil)),
```

Comment:
Can we make it a real annotation by putting it in `d.Spec.Template.Annotations`?